mgmt/rest/v2/metric.go

```diff
@@ -0,0 +1,178 @@
+/*
+http://www.apache.org/licenses/LICENSE-2.0.txt
+
+
+Copyright 2015 Intel Corporation
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v2
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+	"sort"
+	"strconv"
+	"strings"
+
+	"github.com/intelsdi-x/snap/core"
+	"github.com/intelsdi-x/snap/mgmt/rest/v2/response"
+	"github.com/intelsdi-x/snap/pkg/stringutils"
+	"github.com/julienschmidt/httprouter"
+)
+
+func (s *V2) getMetrics(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
+	ver := 0 // 0: get all metrics
+
+	// If we are provided a parameter with the name 'ns' we need to
+	// perform a query
+	q := r.URL.Query()
+	v := q.Get("ver")
+	ns_query := q.Get("ns")
+	if ns_query != "" {
+		ver = 0 // 0: get all versions
+		if v != "" {
+			var err error
+			ver, err = strconv.Atoi(v)
```

Comment:
Doing `ver, err := strconv.Atoi(v)` will create a new variable `ver` that is not in the same context that the original one. Basically it will throw an error "variable unused".